src/main/java/seedu/address/logic/commands/person/DeletePersonCommand.java

```diff
@@ -43,6 +45,20 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
         Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+        boolean finish = false;
+
+        while (!finish) {
+            ObservableList<Appointment> appointments = model.getFilteredAppointmentList();
+            finish = true;
+            for (Appointment a : appointments) {
+                if (personToDelete.getId().equals(a.getPersonId())) {
+                    model.deleteAppointment(a);
+                    finish = false;
```

Comment:
Good practice to ensure that all appointments linked to the person wilil be deleted.